src/games/strategy/engine/EngineVersion.java

```diff
@@ -1,7 +1,33 @@
 package games.strategy.engine;
 
+import games.strategy.engine.config.GameEngineProperty;
+import games.strategy.engine.config.PropertyReader;
 import games.strategy.util.Version;
 
 public class EngineVersion {
-  public static final Version VERSION = new Version(1, 8, 0, 9);
+  private final Version version;
+  private final String exactVersion;
+
+  public EngineVersion(PropertyReader propertyReader) {
+    exactVersion = propertyReader.readProperty(GameEngineProperty.ENGINE_VERSION);
+    version = new Version(exactVersion);
```

Comment:
The existing code works fine to parse the 5 digit version number into a 4 digit version number. It cares if you have no ints, but it doesn't check a max, and it iterates over only the first 4 to create the `Version` number.